cpp/src/strings/search/find_instance.cu

```diff
@@ -63,23 +63,27 @@ CUDF_KERNEL void find_instance_warp_parallel_fn(column_device_view const d_strin
   auto const max_pos   = d_str.size_bytes();
   size_type char_pos   = max_pos;
   size_type char_count = 0;
-  size_type count      = 0;
-  for (auto itr = begin + lane_idx; itr + d_target.size_bytes() <= end;
-       itr += cudf::detail::warp_size) {
-    size_type const is_char = !is_utf8_continuation_char(*itr);
-    size_type const found   = is_char && (d_target.compare(itr, d_target.size_bytes()) == 0);
+  size_type byte_count = 0;
+  size_type offset     = 0;
+  auto itr             = begin + lane_idx;
+  while (byte_count + d_target.size_bytes() <= d_str.size_bytes()) {
+    size_type const is_char =
+      (itr + d_target.size_bytes() <= end) && !is_utf8_continuation_char(*itr);
+    size_type const found = is_char && (d_target.compare(itr, d_target.size_bytes()) == 0);
     // count of threads that matched in this warp and produce an offset in each thread
     auto const found_count = cg::reduce(warp, found, cg::plus<size_type>());
```

Comment:
A cleaner way is to use [cg::binary_partition](https://docs.nvidia.com/cuda/cuda-c-programming-guide/#binary-partition) to generate the active warp and pass that active warp to `cg::reduce`, e.g. https://godbolt.org/z/cKb1nP7x9

Reply:
I see what you mean. In cuco, we normally use `cg::any`  to keep the logic clean and to make sure the whole warp/tile can fall into the loop iteration (see [example](https://github.com/NVIDIA/cuCollections/blob/ad9c478ad174047d3c426e7775d9372aabb0759f/include/cuco/detail/static_multimap/kernels.cuh#L392)), but it's not a requirement and as you pointed out, it requires extra syncs. Let me resolve this.